open-api/rest-catalog-open-api.yaml

```diff
@@ -532,6 +532,100 @@ paths:
         5XX:
           $ref: '#/components/responses/ServerErrorResponse'
 
+  /v1/{prefix}/namespaces/{namespace}/tables/{table}/preplan:
+    parameters:
+      - $ref: '#/components/parameters/prefix'
+      - $ref: '#/components/parameters/namespace'
+      - $ref: '#/components/parameters/table'
+    post:
+      tags:
+        - Catalog API
+      summary: Find plan-tasks based on a query.
+      description:
+        When a user submits a query, this operation will find the relevant plan-tasks
+        based on the user's selected columns, and filters. The plan-tasks can be later used during PlanTable
+        to distribute this work for performance gain.
+      operationId: PreplanTable
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PreplanTableRequest'
+      responses:
+        200:
+          $ref: '#/components/responses/PreplanTableResponse'
+        400:
+          $ref: '#/components/responses/BadRequestErrorResponse'
+        401:
+          $ref: '#/components/responses/UnauthorizedResponse'
+        403:
+          $ref: '#/components/responses/ForbiddenResponse'
+        404:
+          description:
+            Not Found
+            - NoSuchTableException, the table does not exist
+            - NoSuchNamespaceException, the namespace does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+              examples:
+                TableDoesNotExist:
+                  $ref: '#/components/examples/NoSuchTableError'
+                NamespaceDoesNotExist:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+        419:
+          $ref: '#/components/responses/AuthenticationTimeoutResponse'
+        503:
+          $ref: '#/components/responses/ServiceUnavailableResponse'
+        5XX:
+          $ref: '#/components/responses/ServerErrorResponse'
+
+  /v1/{prefix}/namespaces/{namespace}/tables/{table}/plan/:
+    parameters:
+      - $ref: '#/components/parameters/prefix'
+      - $ref: '#/components/parameters/namespace'
+      - $ref: '#/components/parameters/table'
+    post:
+      tags:
+        - Catalog API
+      summary: Initiate plan on table to find relevant file scan tasks for a specific query.
+      operationId: PlanTable
+      description:
+        When a user submits a query, we can do a plan to find the relevant file scan tasks
+        based on the user's selected columns, and filters. Passing the param plan-tasks can be used
+        for enhanced query performance, by distributing the planning work.
```

Comment:
this sentence reads a little weird. maybe `For distributed planning, use plan-tasks to pass in a portion of scan planning`?

Reply:
How about 
```
For enhanced query performance, users can provide a plan-task which will distribute the planning work. 
```

Reply:
I suggested a description update below.